src/util/ping.ts

```diff
@@ -0,0 +1,52 @@
+export const ping = (host: string): Promise<any> => {
+  return new Promise((resolve) => {
+    const socket = new WebSocket(host);
+
+    let pings = 0;
+    const latencies: number[] = [];
+    let pingInterval: any;
+    let timeoutInterval: any;
+
+    const ping = () => {
+      pings++;
+
+      const start = Date.now();
+      socket.send('ping');
+
+      socket.onmessage = (event) => {
+        if (event.data === 'pong') {
+          const latency = Date.now() - start;
+          latencies.push(latency);
+
+          if (pings >= 3) {
+            clearInterval(pingInterval);
+            clearInterval(timeoutInterval);
+            const median = latencies.sort((a, b) => a - b)[1];
+            socket.close();
+            resolve(median);
+          }
+        }
+      };
+    }
+
+    const timeout = () => {
+      console.error('Socket timeout');
```

Comment:
Should make these `Logger.error` (or `Logger.warn`)

Reply:
Good call